tests/test_eval_gauntlet.py

```diff
@@ -106,4 +106,5 @@ def test_gauntlet_callback():
         name = f'icl/metrics/eval_gauntlet/{category}'
         assert result[name] == pytest.approx(0.25)
 
-    assert result['icl/metrics/eval_gauntlet/average'] == pytest.approx(0.25)
```

Comment:
I think it is time to add more unit tests here. This test tests one particular setup, but there are many configurable and optional args to the eval gauntlet. Can you please add more tests that test a variety of the options? I would recommend using `pytest.mark.parametrize` to specify different combinations of the various args to the callback.